Xcode IDE profile

deniskrut opened the pull request:
This is a simple IDE profile for Xcode.

deniskrut committed:
Add Xcode IDE profile

feat: Add Xcode IDE profile

Add basic support for Xcode IDE for ease of development.

deniskrut committed:
Add Xcode IDE Readme

feat: Add Xcode IDE Readme.MD

Describes how to use Xcode IDE profile.

deniskrut committed:
Add note about project generation in README.md

docs: Add note about project generation in README.md

deniskrut committed:
Exclude all xcuserdata directories from git

bug: exclude all xcuserdata directories from git

deniskrut committed:
Add instructions how to get Xcode project that can build

docs: Add instructions how to get Xcode project that can build

If current project is opened on a different machine, it will not be
able to build due to use of absolute paths in the cmake Xcode
generator. Thus suggesting to re-generate the Xcode project in order to
get project that can build.

deniskrut committed:
Exclude build files from repo

bug: Exclude build files from repo

ide_profiles/xcode/Debug/ and ide_profiles/xcode/ExtendedKF.build/
files are generated at the build time and are not required to be in the
repo

cameronwp commented on ide_profiles/xcode/README.md:
```diff
@@ -1,7 +1,12 @@
 
-To generate an Xcode project or update existing one, use `cmake -G "Xcode" ../..` command in the current directory.
+To build using Xcode, re-generate this project as follows:
+
```
nit: could we get rid of this empty line?

deniskrut replied:
It looks like when I remove empty line, markdown displays the document incorrectly. All bullet points are in one line. I don't think we should remove it.

cameronwp commented on ide_profiles/xcode/README.md:
```diff
@@ -1,7 +1,12 @@
+
+1. Remove all files from this directory except of `README.md`
+2. Execute the following command in the current directory: `cmake -G "Xcode" ../..`
+
```
same deal. let's avoid empty lines

deniskrut replied:
Same as above. It will break the markup if I remove it.

cameronwp replied:
hey sorry about that. for whatever reason, I was reading it as a random space, not a newline. my mistake!

cameronwp commented on ide_profiles/xcode/README.md:
```diff
@@ -1,7 +1,12 @@
-To generate an Xcode project or update existing one, use `cmake -G "Xcode" ../..` command in the current directory.
+To build using Xcode, re-generate this project as follows:
+
+1. Remove all files from this directory except of `README.md`
```
if students need to remove all of these files, then let's remove them from of the repo.

cameronwp commented:
Thanks for submitting this PR! I'd like to clean it up a bit before merging. Notably, I'd like any autogenerated files that are specific to your machine (the files you tell students to delete) to be removed. Take a look at my comments and let me know if you have any questions.

cameronwp commented on .gitignore:
```diff
@@ -6,3 +6,19 @@ build/
 .vscode/
+
+# Xcode
+*.pbxuser
```
as I'm not an Xcode user, I'm not sure what all of this is. what are all of these?

deniskrut replied:
These files contain user settings for Xcode. This particular one saves positions of windows. This should not be in the repo.

cameronwp replied:
Cool

deniskrut committed:
Remove all files that are dependent on local path

style: Remove all files that are dependent on local path

Part of the files are used for compilation, and will not work outside
of creator’s project path. Therefore they are being removed. Project
can still be used for editing.

deniskrut commented:
Thanks Cameron! I've removed all build-related files. Please let me know if there is anything else I need to address.

cameronwp commented on ide_profiles/xcode/README.md:
```diff
@@ -0,0 +1,12 @@
+
+To build using Xcode, re-generate this project as follows:
+
+1. Remove all files from this directory except of `README.md`
```
Is this still relevant after removing all the other files?

deniskrut replied:
Yep. But since there is only one file left, I've updated the language to be more clear.

cameronwp commented on ide_profiles/xcode/ExtendedKF.xcodeproj/project.pbxproj:
```diff
@@ -0,0 +1,680 @@
+			buildSettings = {
+				ONLY_ACTIVE_ARCH = YES;
+				SDKROOT = /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk;
+				SYMROOT = "/Users/denyskrut/Documents/Extended-Kalman-Filter/ide_profiles/xcode/build";
```
it looks like this path (and a few others) are specific to your machine. is there a way around this? could students generate this file themselves?

deniskrut replied:
Yes, students can generate these files themselves. However my thought was to still keep these files to make it easier for students to get started. As is project can be used for editing. Students can use instructions in the `README.md` to generate project that can build. Please let me know what you think.

deniskrut committed:
Clarify instructions on how to generate project that can be built

docs: Clarify instructions on how to generate project that can be built

Use easier language on what is required to be removed.

deniskrut commented:
Hey @cameronwp, I've updated the pull request and answered on the comments. Please let me know if there is something else that I can do.

deniskrut commented:
Hi @cameronwp! Thank you for your feedback so far. Do you have any further comments? Would it be preferable if I remove all files except of `README.md` and provide instructions how to generate the project, build it and debug it? Thanks!

cameronwp commented:
Hey sorry for the silence. I've been trying to find someone at Udacity with more Xcode experience than I do to help me review :)

I think your last idea, removing everything except the README w/generation instructions, is great! That's a perfect starting point for helping other folks get up and running. We can always add configuration files later if they seem necessary.

deniskrut committed:
Instructions how to generate and use Xcode project

docs: Instructions how to generate and use Xcode project

In detail description on how to generate, configure for debugging and
use the Xcode profile.

deniskrut committed:
Remove preconfigured project

feat: Remove preconfigured project

We have instruction on how to generate project, so pre-generated
project is not essential any more.

deniskrut committed:
Fix images in the documentation for Xcode IDE profile

docs: Fix images in the documentation for Xcode IDE profile

deniskrut committed:
Remove duplicate instructions from Xcode IDE profile readme

docs: Remove duplicate instructions from Xcode IDE profile readme

deniskrut commented:
Hey @cameronwp, I've removed the Xcode profile and added instruction on how to generate one. Please check it out!

deniskrut committed:
Correct title size for Xcode readme

docs: Correct title size for Xcode readme

deniskrut committed:
Use unicode symbols to represent "Run" button for Xcode readme

docs: Use unicode symbols to represent "Run" button for Xcode readme

deniskrut committed:
Clarify how how to output debug data in Xcode

docs: Clarify how how to output debug data in Xcode

cameronwp commented:
lgtm! thanks for all your work here
